projects/packages/my-jetpack/_inc/components/product-interstitial/index.jsx

```diff
@@ -40,7 +46,39 @@ export default function ProductInterstitial( { slug, children = null } ) {
 	}, [ recordEvent, slug ] );
 
 	const Product = isUpgradableByBundle ? ProductDetailCard : ProductDetail;
+	const { isUserConnected } = useMyJetpackConnection();
+
+	const addProductUrl = isFree
+		? null
+		: getProductCheckoutUrl( `jetpack_${ slug }`, isUserConnected ); // @ToDo: Remove this when we have a new product structure.
+
+	const navigateToMyJetpackOverviewPage = useMyJetpackNavigate( '/' );
+	const navigateToCheckoutPage = useCallback( () => {
+		window.location.href = addProductUrl;
+	}, [ addProductUrl ] );
+
+	const afterInstallation = useCallback(
+		free => {
+			if ( free ) {
+				navigateToMyJetpackOverviewPage();
+			} else {
+				navigateToCheckoutPage();
+			}
+		},
+		[ navigateToMyJetpackOverviewPage, navigateToCheckoutPage ]
+	);
 
+	const clickHandler = useCallback( () => {
+		if ( installsPlugin ) {
+			activate()
+				.then( () => {
+					afterInstallation( isFree );
+				} )
+				.catch( () => {
+					afterInstallation( isFree );
+				} );
```

Comment:
minor suggestion.
Maybe you can use [.finally()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally) here to void duplicate the same call for `then()` and `catch()`.